src/lib.rs

```diff
@@ -39,6 +39,22 @@ pub fn find_app_by_id<'a>(
         .find(|entry| entry.matches_id(app_id))
         // Only if that fails should we fall back to searching by name
         .or_else(|| entries.iter().find(|entry| entry.matches_name(app_id)))
+        // Or match by exact exec name
+        .or_else(|| {
+            entries
+                .iter()
+                .find(|entry| entry.exec().is_some_and(|exec| exec == app_id))
+        })
+        // Or any app with the exec in its arguments
+        .or_else(|| {
+            entries.iter().find(|entry| {
+                entry.exec().is_some_and(|exec| {
+                    exec.split_ascii_whitespace()
+                        .next()
+                        .is_some_and(|exec| exec == app_id)
+                })
+            })
+        })
```

Comment:
These (and see comment below) are pretty much not in spec. Apps posing as other applications is a legit security risk (+ is it super weird if the launcher matches the wrong app, we had these issues before with the weird heuristic approach).

Does any other DE do this? If we have apps that are correctly detected in other DEs we should compare with their detection routines.